engine/transform.go

```diff
@@ -73,6 +73,21 @@ func addDefaultHaveAllChecksRunCompleted(str string) string {
 	return r.ReplaceAllString(str, `$$haveAllChecksRunCompleted([], "")`)
 }
 
+func addDefaultJoinSeparator(str string) string {
+	r := regexp.MustCompile(`\$join\(((\[.*\])|(\$(team|group)\("[^"]*"\))|(\$assignees\(\)))(,\s*"([^"]*)")?\)`)
```

Comment:
What's the reasoning behind `team`, `group`, and `assignees`? I may be missing something 😅

Isn't a simpler regex enough?
```suggestion
	r := regexp.MustCompile(`\$join\((\[.*\]|([^,]*))(,\s*"([^"]*)")?\)`)
```

If we have the literal values `team`, `group`, and `assignees`, we need to keep updating this transform every time there's a new built-in that returns a []string. For instance, [$filePaths()](https://github.com/reviewpad/reviewpad/issues/538).

Reply:
Thanks for pointing this out! Simplified the regex.